topic-operator/src/main/java/io/strimzi/operator/topic/KafkaStreamsConfiguration.java

```diff
@@ -0,0 +1,215 @@
+import io.apicurio.registry.streams.distore.KeyValueStoreGrpcImplLocalDispatcher;
+import io.apicurio.registry.streams.distore.UnknownStatusDescriptionInterceptor;
+import io.apicurio.registry.streams.distore.proto.KeyValueStoreGrpc;
+import io.apicurio.registry.streams.utils.ForeachActionDispatcher;
+import io.apicurio.registry.streams.utils.Lifecycle;
+import io.apicurio.registry.streams.utils.LoggingStateRestoreListener;
+import io.apicurio.registry.utils.ConcurrentUtil;
+import io.apicurio.registry.utils.kafka.AsyncProducer;
+import io.apicurio.registry.utils.kafka.ProducerActions;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.ServerInterceptors;
+import io.grpc.Status;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.Topology;
+import org.apache.kafka.streams.errors.InvalidStateStoreException;
+import org.apache.kafka.streams.state.HostInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Configuration required for KafkaStreamsTopicStore
+ */
+public class KafkaStreamsConfiguration {
+    private static final Logger log = LoggerFactory.getLogger(KafkaStreamsConfiguration.class);
+
+    private final List<AutoCloseable> closeables = new ArrayList<>();
+
+    /* test */ KafkaStreams streams;
+    private TopicStore topicStore;
+
+    public void start(Config config, Properties kafkaProperties) {
+        try {
+            String storeTopic = config.get(Config.STORE_TOPIC);
+            String storeName = config.get(Config.STORE_NAME);
+
+            long timeoutMillis = config.get(Config.STALE_RESULT_TIMEOUT);
+            ForeachActionDispatcher<String, Integer> dispatcher = new ForeachActionDispatcher<>();
+            WaitForResultService serviceImpl = new WaitForResultService(timeoutMillis, dispatcher);
+            closeables.add(serviceImpl);
+
+            Topology topology = new TopicStoreTopologyProvider(storeTopic, storeName, kafkaProperties, dispatcher).get();
+            streams = new KafkaStreams(topology, kafkaProperties);
+            streams.setGlobalStateRestoreListener(new LoggingStateRestoreListener());
+            closeables.add(streams);
+            streams.start();
+
+            String appServer = config.get(Config.APPLICATION_SERVER);
+            String[] hostPort = appServer.split(":");
+            log.info("Application server gRPC: '{}'", appServer);
+            HostInfo hostInfo = new HostInfo(hostPort[0], Integer.parseInt(hostPort[1]));
+
+            FilterPredicate<String, Topic> filter = (s, s1, s2, topic) -> true;
+
+            DistributedReadOnlyKeyValueStore<String, Topic> store = new DistributedReadOnlyKeyValueStore<>(
```

Comment:
As discussed elsewhere, we don't really need the distributed aspect of this. If there some other implementation which would avoid the need for the GRPC layer?

Reply:
If it's only ever gonna be single instance, then we're good with Streams already provide -- some sort of in-memory KVStore.

Otoh, I would somehow leave my current distributed code -- so it doesn't go to waste,
or if someone needs more then one instance.
How do you suggest to do this?